authors/apps/bookmark/tests/test_bookmark.py

```diff
@@ -0,0 +1,146 @@
+            HTTP_AUTHORIZATION='Token ' + self.access_token
+        )
+
+        self.assertIn(response.data['message'], 'Article {data} was not found'.format(data=slug))
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertIn(response2.data['message'], 'Article {data} was not found'.format(data=slug))
+        self.assertEqual(response2.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_bookmark_model(self):
+        """"
+        test model can create user profile upon successful sign up
+        """
+
+        initial_count = BookmarkArticle.objects.count()
+        res2 = self.post_article(self.article)
+        slug = res2.data['slug']
+        
+        self.client.post(
+            '/api/article/{slug}/bookmark'.format(slug=slug),
+            content_type='application/json',
+            HTTP_AUTHORIZATION='Token ' + self.access_token
+        )
+        new_count = BookmarkArticle.objects.count()
+        self.assertNotEqual(initial_count, new_count)
+
+    def test_user_can_bookmark_article(self):
+        """
+        Test user can bookmark an article
+        """
+
+        res2 = self.post_article(self.article)
+        slug = res2.data['slug']
+
+        response = self.client.post(
+            '/api/article/{slug}/bookmark'.format(slug=slug),
+            content_type='application/json',
+            HTTP_AUTHORIZATION='Token ' + self.access_token
+        )
+
+        response2 = self.client.post(
+            '/api/article/{slug}/bookmark'.format(slug=slug),
+            content_type='application/json',
+            HTTP_AUTHORIZATION='Token ' + self.access_token
+        )
+
+        self.assertIn(response.data['message'], 'Bookmarking article {data} was successful'.format(data=slug))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn(response2.data['message'], 'You have already bookmarked this article')
+        self.assertEqual(response2.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_user_can_unbookmark_article(self):
+        """
+        Test user can unfbookmark an article
+        """
+
+        res2 = self.post_article(self.article)
+        slug = res2.data['slug']
+
+        self.client.post(
+            '/api/article/{slug}/bookmark'.format(slug=slug),
+            content_type='application/json',
+            HTTP_AUTHORIZATION='Token ' + self.access_token
+        )
+        response = self.client.delete(
+            '/api/article/{slug}/bookmark'.format(slug=slug),
+            content_type='application/json',
+            HTTP_AUTHORIZATION='Token ' + self.access_token
+        )
+
+        response2 = self.client.delete(
+            '/api/article/{slug}/bookmark'.format(slug=slug),
+            content_type='application/json',
+            HTTP_AUTHORIZATION='Token ' + self.access_token
+        )
+
+        self.assertIn(response.data['message'], 'Unbookmarking article {data} was successful'.format(data=slug))
```

Comment:
Define a function for these sets of assert statements. This is for good DRY practice.

Reply:
@bettblake08 I have tried to create a function but it does not simplify the code.